backup/data.go

```diff
@@ -110,6 +112,159 @@ func BackupSingleTableData(table Table, rowsCopiedMap map[uint32]int64, counters
 	return nil
 }
 
+// backupDataForAllTablesCopyQueue does not backup tables in parallel. This is
+// specifically for single-data-file. While tables backed up one at a time, it
+// is possible to save a significant amount of time by queuing up the next
+// table to copy. Worker 0 does not have tables pre-assigned to it. The copy
+// queue uses worker 0 a special deferred worker in the event that the other
+// workers encounter locking issues. Worker 0 already has all locks on the
+// tables so it will not run into locking issues.
+func backupDataForAllTablesCopyQueue(tables []Table) []map[uint32]int64 {
+	var numExtOrForeignTables int64
+	for _, table := range tables {
+		if table.SkipDataBackup() {
```

Comment:
@hughcapet, I think we had filtered all external tables earlier when we found themselves here.